### PR TITLE
Model relations

### PR DESCRIPTION
**Pull request by HasanbekElmurodov**
model relations add

### CODE REVIEW

**Comment by makhsam** [comment]
@Mirvali97 @Nurxoniddin tekshirib berila

**Comment by makhsam** [comment]
Yigitlar kim VS Code ishlatsa mana shu php intelligence ni o'rnatib olila:

PHP Intelephense - Visual Studio Marketplace
https://marketplace.visualstudio.com/items?itemName=bmewburn.vscode-intelephense-client

File: app/Models/University.php [hunk 1]
```diff
@@ -8,4 +8,11 @@
 class University extends Model
 {
     use HasFactory;
+
```
Comment (Mirvali97) [inline app/Models/University.php:11]: @makhsam  Abuturentlarda bir nechata univerticy buladimi yoke Univercityni bir necha talabalari deb ishlatilganmi Tushinaolmadik

File: app/Models/UserExam.php [hunk 2]
```diff
@@ -8,4 +8,14 @@
 class UserExam extends Model
 {
     use HasFactory;
+
+    public function userTests(){
+        return $this->hasMany(UserTest::class);
+    }
+    public function users(){
+        return $this->hasMany(User::class);
```
Comment (Mirvali97) [inline app/Models/UserExam.php:16]: @makhsam  belong-to  shekille

File: app/Models/UserTest.php [hunk 3]
```diff
@@ -8,4 +8,17 @@
 class UserTest extends Model
 {
     use HasFactory;
+
+    public function subject(){
```
Comment (Mirvali97) [inline app/Models/UserTest.php:12]: @makhsam  @HasanbekElmurodov  belongs-to

**Review by Mirvali97 — COMMENTED** [review]
checked-13, pending 4  @makhsam @Nurxoniddin @HasanbekElmurodov  @Dilshod0609

File: app/Models/UserTest.php [hunk 4]
```diff
@@ -8,4 +8,17 @@
...
+    public function testAnswer(){
+        return $this->hasOne(UserTestAnswer::class);
+    }
+    public function userExam(){
+        return $this->belongsTo(UserExam::class);
+    }
+    public function users(){
+        return $this->hasMany(User::class);
+    }
 }
```
Comment (makhsam) [inline app/Models/UserTest.php:24]: @HasanbekElmurodov `UserTest` modelda `userExam()` dan tashqari hamma relationshipla xato ketibdi. Iltimos boshqatdan o'ziz tekshirib chiqin

File: app/Models/University.php [hunk 5]
```diff
@@ -8,4 +8,11 @@
...
+    public function users(){
+        return $this->hasMany(User::class);
+    }
```
Comment (makhsam) [inline app/Models/University.php:14]: University modelning User bilan aloqasi yo'q. Bu yerda users() relationship noo'rin

File: app/Models/UserExam.php (hunk 2 above)
Comment (makhsam) [reply app/Models/UserExam.php:16]: To'g'ri @Mirvali97 . Qaysi tableda `user_id` column bo'lsa, o'sha modelda `user() => belongsTo` relationship bo'ladi. HasMany ham, HasOne ham emas, boshqa joylarda ham shu xatolik bor.

**Review by makhsam — COMMENTED** [review]
@HasanbekElmurodov hamma modellani o'zizam boshqatdan tekshirib chiqin

**Comment by HasanbekElmurodov** [comment]
@makhsam aka relationlarni to'g'irladim va factory seederlarni ham push qilvordim qo'shib. Tekshirib berila

**Comment by makhsam** [comment]
@Mirvali97 tekshirib berin. O'ziz `model_relations` branchiga o'tib turib, bitta tekshirib chiqin. Keyin code formatting'niyam to'g'rilab bersez zo'r bo'lardi 😎

**Comment by Mirvali97** [comment]
> @Mirvali97 tekshirib berin. O'ziz `model_relations` branchiga o'tib turib, bitta tekshirib chiqin. Keyin code formatting'niyam to'g'rilab bersez zo'r bo'lardi

@makhsam  Hop Faqat code formattingni alohida branchga pull request qilaman

**Review by Mirvali97 — COMMENTED** [review]
viewed all  chenged files

**Review by makhsam — APPROVED** [review]
All checks have passed

**Pull request merged**